CRM/Core/Payment/BaseIPN.php

```diff
@@ -92,9 +92,7 @@ public function validateData($input, &$ids, &$objects, $required = TRUE, $paymen
     $contribution = new CRM_Contribute_BAO_Contribution();
     $contribution->id = $ids['contribution'];
     if (!$contribution->find(TRUE)) {
-      CRM_Core_Error::debug_log_message("Could not find contribution record: {$contribution->id} in IPN request: " . print_r($input, TRUE));
-      echo "Failure: Could not find contribution record for {$contribution->id}<p>";
-      return FALSE;
+      throw new CRM_Core_Exception('Failure: Could not find contribution record for ' . (int) $contribution->id, NULL, ['context' => "Could not find contribution record: {$contribution->id} in IPN request: " . print_r($input, TRUE)]);
```

Comment:
@eileenmcnaughton I would note that I think this will cause HTTP 500 errors to be thrown rather than HTTP 200 back to the IPN

Reply:
@seamuslee001 - except you already merged the try-catch to wrap around it

Reply:
ah ok I guess so right